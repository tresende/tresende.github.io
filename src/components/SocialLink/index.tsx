import React from 'react'

export type SocialLinkProps = {
  name: 'linkedin' | 'email' | 'github' | 'facebook' | 'instagram'
  url: string
}

const SocialLink = ({ name, url }: SocialLinkProps) => {
  return (
    <a href={url} target="_blank" rel="noreferrer">
      <img src={`/img/${name}-min.png`} alt={`${name} icon`} />
    </a>
  )
}
export default SocialLink
