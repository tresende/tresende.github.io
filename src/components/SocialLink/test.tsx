import { render, screen } from '@testing-library/react'

import SocialLink from '.'

const mock = {
  githubUrl: 'github.com/tresende',
  icon: '/img/github.png'
}

describe('<SocialLink />', () => {
  it('should render social link', () => {
    const { container } = render(<SocialLink name="github" url="github.com/tresende" />)

    expect(screen.getByRole('link')).toHaveAttribute('href', mock.githubUrl)
    expect(screen.getByRole('img')).toHaveAttribute('src', mock.icon)
    expect(container).toMatchSnapshot()
  })
})
