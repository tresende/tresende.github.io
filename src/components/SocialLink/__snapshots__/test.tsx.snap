// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialLink /> should render social link 1`] = `
<div>
  <a
    href="github.com/tresende"
    rel="noreferrer"
    target="_blank"
  >
    <img
      alt="github icon"
      height="32"
      src="/img/github.png"
      width="32"
    />
  </a>
</div>
`;
